src/main/java/it/gov/pagopa/pu/pagopapayments/service/aca/AcaService.java

```diff
@@ -28,22 +29,22 @@ public AcaService(AcaClient acaClient, AcaDebtPositionMapper acaDebtPositionMapp
     this.brokerService = brokerService;
   }
 
-  public void create(DebtPositionDTO debtPosition, String accessToken) {
-    invokePaCreatePositionImpl(debtPosition, OPERATION.CREATE, accessToken);
+  public List<String> create(DebtPositionDTO debtPosition, String accessToken) {
+    return invokePaCreatePositionImpl(debtPosition, OPERATION.CREATE, accessToken);
   }
 
-  public void update(DebtPositionDTO debtPosition, String accessToken) {
-    invokePaCreatePositionImpl(debtPosition, OPERATION.UPDATE, accessToken);
+  public List<String> update(DebtPositionDTO debtPosition, String accessToken) {
+    return invokePaCreatePositionImpl(debtPosition, OPERATION.UPDATE, accessToken);
   }
 
-  public void delete(DebtPositionDTO debtPosition, String accessToken) {
-    invokePaCreatePositionImpl(debtPosition, OPERATION.DELETE, accessToken);
+  public List<String> delete(DebtPositionDTO debtPosition, String accessToken) {
+    return invokePaCreatePositionImpl(debtPosition, OPERATION.DELETE, accessToken);
   }
 
-  private void invokePaCreatePositionImpl(DebtPositionDTO debtPosition, OPERATION operation, String accessToken) {
-    List<NewDebtPositionRequest> debtPostionToSendACA = acaDebtPositionMapper.mapToNewDebtPositionRequest(debtPosition);
+  private List<String> invokePaCreatePositionImpl(DebtPositionDTO debtPosition, OPERATION operation, String accessToken) {
+    List<Pair<String, NewDebtPositionRequest>> debtPostionToSendACA = acaDebtPositionMapper.mapToNewDebtPositionRequest(debtPosition);
     Pair<BrokerApiKeys, String> brokerData = brokerService.getBrokerApiKeyAndSegregationCodesByOrganizationId(debtPosition.getOrganizationId(), accessToken);
-    debtPostionToSendACA.forEach(newDebtPositionRequest -> {
+    debtPostionToSendACA.stream().map(Pair::getRight).forEach(newDebtPositionRequest -> {
```

Comment:
don't cycle twice the input list
```suggestion
    return debtPostionToSendACA.stream().forEach(p -> {
       NewDebtPositionRequest newDebtPositionRequest = p.getRight();
       ....
       return p.getLeft();
```

Reply:
done (don't cycle twice the list), but a bit differently. I guest that in your suggestion you mean "stream.map()" and not "stream.forEach()" (that returns void). But I don't like very much to put in the map() something that may have side effect or change state, so I just explicitelly fill a List in the forEach() loop.

Reply:
it's better the .map operation and finalize the stream with a toList()
This way you will create in a more efficient manner the resulted list

Reply:
done